plaidml/op/op.h

```diff
@@ -337,6 +337,64 @@ inline edsl::Tensor minimum(const edsl::Tensor& X, const edsl::Tensor& Y) {
   return details::op("minimum", args).as_tensor();
 }
 
+class mvn {
+ public:
+  explicit mvn(const plaidml::edsl::Tensor& I)
+      : I_(I),
+        axes_(edsl::None()),
+        normalize_variance_(true),
+        epsilon_(1e-9),
+        across_channels_(true),
+        layout_("NHWC") {}
+
+  mvn& axes(const std::vector<int64_t>& axes) {
+    // negative axes interpreted as in numpy
+    axes_ = edsl::make_tuple(axes);
+    return *this;
+  }
+
+  mvn& remove_axes_parameter() {
+    // Restores axes to the default of being unused (meaning use `across_channels` & `layout` instead)
+    axes_ = edsl::None();
+    return *this;
+  }
+
+  mvn& across_channels(bool flag) {
+    // only used if `axes` is `None`
+    across_channels_ = flag;
+    return *this;
+  }
+
+  mvn& normalize_variance(bool flag) {
+    normalize_variance_ = flag;
+    return *this;
+  }
+
+  mvn& epsilon(double value) {
+    epsilon_ = value;
+    return *this;
+  }
+
+  mvn& layout(const std::string& value) {
+    // only used if `axes` is `None`
```

Comment:
Maybe we could add a check here and throw if the user specified `axes`. And also we could do the same for `axes` (if `layout` was set, then throw).